lib/Blog/BlogPostRepository.php

```diff
@@ -32,6 +33,16 @@ public function find(SourceFile $sourceFile) : BlogPost
         );
     }
 
+    /**
+     * @return BlogPost[]
+     */
+    public function findPaginated(int $page = 1, int $perPage = 10) : array
```

Comment:
How should this behave on values `< 1`? Throwing an exception? Can we use `assert` here?

Reply:
Hmm, good question. I am not sure. Do you think we should do something?

Reply:
`array_slice` allows negative values for offset and length, but that's an implementation detail, that is not known to the code invoking `findPaginated()`. Are negative numbers allowed by the repository method? If not, then we should check them.

Reply:
I get what you are saying. Do you think it is worth it in such a simple application where we are the ones calling findPaginated in 2 places?

Reply:
Yes